README.md

```diff
@@ -18,6 +18,11 @@ Prerequisites
 
 This requires Go 1.4 or above.
 
+Design Constraints
+------------------
+
+The `google.golang.org/grpc` Go package must not depend on any external Go libraries. This is to simplify use when vendored into other Go projects.
```

Comment:
Well, I can't define external because I don't know the rules. Clearly we should document the rules so we can point to the rules when people open PRs and hopefully avoid duplicate work.